spec/models/generic_object_spec.rb

```diff
@@ -0,0 +1,80 @@
+describe GenericObject do
+  let(:go_object_name) { "load_balancer_1" }
+  let(:maxNumber)      { 100 }
+  let(:server_name)    { "test_server" }
+  let(:data_read)      { 345.67 }
+  let(:s_time)         { Time.now.utc }
+
+  let(:definition) do
+    FactoryGirl.create(
+      :generic_object_definition,
+      :name       => "test_definition",
+      :properties => {
+        :attributes => {
+          :maxNumber => "integer",
+          :server    => "string",
+          :flag      => "boolean",
+          :data_read => "float",
+          :s_time    => "datetime"
+        }
+      }
+    )
+  end
+
+  let(:go) do
+    GenericObject.new(
+      definition.name,
+      :name      => go_object_name,
+      :maxNumber => maxNumber,
+      :server    => server_name,
+      :flag      => true,
+      :data_read => data_read,
+      :s_time    => s_time
+    )
+  end
+
+  it "should ensure presence of name" do
+    o = GenericObject.new(definition.name, :name => nil)
+    expect(o.valid?).to be_falsey
+  end
+
+  it "should ensure uniqueness of name" do
+    GenericObject.new(definition.name, :name => "MyGenericObject").save
+    expect(GenericObject.new(definition.name, :name => "MyGenericObject").valid?).to be_falsey
+  end
+
+  context "custom attributes" do
+    it "can be accessed as an attribute" do
+      go.save
+      expect(go.generic_object_definition).to eq(definition)
+      expect(go.name).to                      eq(go_object_name)
+      expect(go.maxNumber).to                 eq(maxNumber)
+      expect(go.server).to                    eq(server_name)
+      expect(go.flag).to                      be_truthy
+      expect(go.data_read).to                 eq(data_read)
+      expect(go.s_time.to_i).to               eq(s_time.to_i)
+    end
+
+    it "can be set as an attribute" do
+      go.maxNumber = maxNumber + 100
+      go.save
```

Comment:
In tests, it might be better to use `save!`, or in the case of this specific tests `update_attributes!` will work as well and trim it up a bit.